data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostRegisterValidator.java

```diff
@@ -0,0 +1,175 @@
+      tags = {TAG_VALIDATOR, TAG_VALIDATOR_REQUIRED},
+      requestBody =
+          @OpenApiRequestBody(
+              content = {@OpenApiContent(from = SignedValidatorRegistrationOpenApiSchema[].class)}),
+      description =
+          "Prepares the beacon node for potential proposers by supplying information required when proposing blocks for the given validators. The information supplied for each validator index is considered persistent until overwritten by new information for the given validator index, or until the beacon node restarts.\n\n"
+              + "Note that because the information is not persistent across beacon node restarts it is recommended that either the beacon node is monitored for restarts or this information is refreshed by resending this request periodically (for example, each epoch).\n\n"
+              + "Also note that requests containing currently inactive or unknown validator indices will be accepted, as they may become active at a later epoch.",
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            description = "Preparation information has been received."),
+        @OpenApiResponse(status = RES_BAD_REQUEST, description = "Invalid parameter supplied."),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR, description = "Beacon node internal error.")
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(final RestApiRequest request) throws JsonProcessingException {
+    request.respondAsync(
+        validatorDataProvider
+            .registerValidators(request.getRequestBody())
+            .thenApply(AsyncApiResponse::respondOk));
+  }
+
+  private static DeserializableTypeDefinition<SszList<SignedValidatorRegistration>> getRequestType(
+      final SchemaDefinitionCache schemaDefinitionCache) {
+    return schemaDefinitionCache
+        .getSchemaDefinition(SpecMilestone.BELLATRIX)
+        .toVersionBellatrix()
+        .orElseThrow()
+        .getSignedValidatorRegistrationsSchema()
+        .getJsonTypeDefinition();
+  }
+
+  @SuppressWarnings("JavaCase")
+  public static class SignedValidatorRegistrationOpenApiSchema {
+    @JsonProperty("message")
+    ValidatorRegistration message;
+
+    @JsonProperty("signature")
+    @Schema(type = "string", format = "byte", description = DESCRIPTION_BYTES96)
+    BLSSignature signature;
+
+    public static class ValidatorRegistration {
+
+      @JsonProperty("fee_recipient")
+      @Schema(
+          type = "string",
+          pattern = PATTERN_EXECUTION_ADDRESS,
+          example = EXAMPLE_EXECUTION_ADDRESS,
+          description = DESCRIPTION_EXECUTION_ADDRESS)
+      Eth1Address fee_recipient;
+
+      @JsonProperty("gas_limit")
+      @Schema(type = "string", format = "uint64", example = EXAMPLE_UINT64)
+      UInt64 gas_limit;
```

Comment:
most of the time schema objects have these as 'public', the reason for that is then you don't also need to do the `@JsonProperty` annotation (or you need a getter, one of the 3 options)

Reply:
seems to be the most compact version :)